include/wil/registry.h

```diff
@@ -17,16 +17,3139 @@
 
 #include <winreg.h>
 #include <new.h> // new(std::nothrow)
-#include "resource.h" // unique_hkey
+#include "registry_helpers.h"
+#include "resource.h"
+
+#ifndef __WIL_WINREG_
+#error This is required for ::wil::unique_hkey support

```

Comment:
What would be the steps to fix this? Can you add that to the error message?

Reply:
ok. I can walk backwards in the wil headers to find all prerequisites.

Reply:
I'm not necessarily saying that---I just mean, if the solution is "hey, you gotta include `foo.h`", then the error message should be `registry.h requires foo.h`, which is an actionalbe message.

Reply:
This new error message is better: is the requirement "registry APIs available" + "resource.h" available?

(And you left off that last bit because we explicitly include resource.h?)